ui-core/src/components/inputs/InputWithSuggestions/InputWithSuggestions.tsx

```diff
@@ -0,0 +1,177 @@
+  value,
+  small,
+  ...inputProps
+}: InputWithSuggestionsProps<T>) => {
+  const [filteredSuggestions, setFilteredSuggestions] = useState<T[]>([]);
+  const [activeSuggestionIndex, setActiveSuggestionIndex] = useState(-1);
+  const [inputValue, setInputValue] = useState(value ? getSuggestionLabel(value) : '');
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const showSuggestions = filteredSuggestions.length > 0;
+
+  const focusInput = () => inputRef.current?.focus();
+
+  const clearInput = () => {
+    setInputValue('');
+    inputRef.current?.focus();
+  };
+
+  const icons = [
+    ...(inputValue
+      ? [
+          {
+            icon: <X size={small ? 'sm' : 'lg'} />,
+            action: clearInput,
+            className: 'clear-icon',
+          },
+        ]
+      : []), // Conditionally include the clear icon only when inputValue is not empty
+    ...(suggestions.length > 0
+      ? [
+          {
+            icon: <ChevronDown size={small ? 'sm' : 'lg'} />,
+            action: focusInput,
+            className: 'chevron-icon',
+          },
+        ]
+      : []), // Conditionally include the chevron icon only when suggestions are not empty
+  ];
+
+  useEffect(() => {
+    if (value) {
+      setInputValue(getSuggestionLabel(value));
+    }
+  }, [value, getSuggestionLabel]);
+
+  const handleInputChange: ChangeEventHandler<HTMLInputElement> = (e) => {
+    const userInput = normalizeString(e.currentTarget.value).trim();
+    setInputValue(e.currentTarget.value);
+
+    if (userInput.trim() === '') {
+      setFilteredSuggestions([]);
+      return;
+    }
+
+    const filtered = suggestions.filter((suggestion) => {
+      const suggestionLabel = normalizeString(getSuggestionLabel(suggestion).toLowerCase());
+      return exactSearch
+        ? suggestionLabel.startsWith(userInput.toLowerCase())
+        : suggestionLabel.includes(userInput.toLowerCase());
+    });
+    setFilteredSuggestions(filtered);
+  };
+
+  const handleKeyDown: KeyboardEventHandler<HTMLInputElement> = (e) => {
+    if (e.key === 'ArrowDown') {
+      setActiveSuggestionIndex((prev) => (prev < filteredSuggestions.length - 1 ? prev + 1 : prev));
+    } else if (e.key === 'ArrowUp') {
+      setActiveSuggestionIndex((prev) => (prev > 0 ? prev - 1 : prev));
+    } else if ((e.key === 'Enter' || e.key === 'Tab') && activeSuggestionIndex >= 0) {
+      selectSuggestion(activeSuggestionIndex);
+    } else if (e.key === 'Escape') {
+      setFilteredSuggestions([]);
+    }
+  };
+
+  const selectSuggestion = (index: number) => {
+    const selectedSuggestion = filteredSuggestions[index];
+    setInputValue(getSuggestionLabel(selectedSuggestion));
+    onChange(selectedSuggestion);
+    setFilteredSuggestions([]);
+    setActiveSuggestionIndex(-1);
+  };
+
+  const sortedSuggestions = useMemo(() => {
+    return suggestions.sort((a, b) => getSuggestionLabel(a).localeCompare(getSuggestionLabel(b)));
+  }, [suggestions, getSuggestionLabel]);
```

Comment:
Strange that eslint doesn't trigger a warning here. I thought we put the same config as osrd.

Also, could you : 
- move this useMemo at the beginning of the component (so we don't miss it)
- replace all uses of `suggestions` in the component by `sortedSuggestions`. I think it might be dangerous to manipulate both const in the components.

You can make these fixes and clean your commit history at the same time if you want.

```suggestion
  const sortedSuggestions = useMemo(
    () => suggestions.sort((a, b) => getSuggestionLabel(a).localeCompare(getSuggestionLabel(b))),
    [suggestions, getSuggestionLabel]
  );
```

Reply:
Nice hint, Fixed :)